### PR TITLE
[VPU] update config header

### PR DESCRIPTION
**Pull request by KorA6**
### Tickets:
 - -74515
 - -77708

### CODE REVIEW

**Comment by KorA6** [comment]
@apankratovantonp , @AntonDudchenko , @DariaMityagina 
Could you please take a look?
@Maxim-Doronin 
FYI

File: src/inference/include/openvino/runtime/intel_myriad/hddl_properties.hpp [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+ * {
+ *     "bypass_device_number": 5
+ * }
+ * It means that 5 device will be used for Bypass scheduler.
+ */
+static constexpr Property<std::string, PropertyMutability::RW> setDeviceTag{"HDDL_SET_DEVICE_TAG"};
+
+/**
+ * @brief [Only for HDDLPlugin]
+ * Type: "YES/NO", default is "NO".
```
Comment (apankratovantonp) [inline src/inference/include/openvino/runtime/intel_myriad/hddl_properties.hpp:117]: Please fix documentation for `bool` types.

File: src/inference/include/openvino/runtime/intel_myriad/myriad_properties.hpp [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+
+/**
+ * @brief Optimize vpu plugin execution to maximize throughput.
+ * This option should be used with integer value which is the requested number of streams.
+ * The only possible values are:
+ *     1
+ *     2
+ *     3
+ */
+static constexpr Property<unsigned int, PropertyMutability::RW> myriadThroughputStreams{"MYRIAD_THROUGHPUT_STREAMS"};
```
Comment (apankratovantonp) [inline src/inference/include/openvino/runtime/intel_myriad/myriad_properties.hpp:127]: try to use `ov::streams::num` property and `ov::streams::AUTO` predefined value

File: src/inference/include/openvino/runtime/intel_myriad/hddl_properties.hpp (hunk 1 above)
Comment (KorA6) [reply src/inference/include/openvino/runtime/intel_myriad/hddl_properties.hpp]: Fixed

File: src/inference/include/openvino/runtime/intel_myriad/myriad_properties.hpp (hunk 2 above)
Comment (KorA6) [reply src/inference/include/openvino/runtime/intel_myriad/myriad_properties.hpp]: Fixed

File: src/inference/include/openvino/runtime/intel_myriad/hddl_properties.hpp [hunk 3]
```diff
@@ -0,0 +1,157 @@
...
+#pragma once
+
+#include <openvino/runtime/properties.hpp>
+
+namespace ov {
+namespace intel_myriad {
+namespace hddl {
+//    RO properties
+/**
+ * @brief Metric to get a int of the device number, String value is METRIC_HDDL_DEVICE_NUM
```
Comment (ilya-lavrenov) [inline src/inference/include/openvino/runtime/intel_myriad/hddl_properties.hpp:14]: metric => property

Comment (ilya-lavrenov) [reply src/inference/include/openvino/runtime/intel_myriad/hddl_properties.hpp:14]: `String value is` comments can be removed everywhere

File: src/inference/include/openvino/runtime/intel_myriad/hddl_properties.hpp [hunk 4]
```diff
@@ -0,0 +1,157 @@
...
+
+/**
+ * @brief  Metric to get a int number of device be using for group, String value is METRIC_HDDL_DEVICE_GROUP_USING_NUM
+ */
+static constexpr Property<int, PropertyMutability::RO> deviceGroupUsingNum{"HDDL_DEVICE_GROUP_USING_NUM"};
+
+/**
+ * @brief  Metric to get a int number of total device, String value is METRIC_HDDL_DEVICE_TOTAL_NUM
+ */
+static constexpr Property<int, PropertyMutability::RO> deviceTotalNum{"HDDL_DEVICE_TOTAL_NUM"};
```
Comment (ilya-lavrenov) [inline src/inference/include/openvino/runtime/intel_myriad/hddl_properties.hpp:78]: @ilyachur am I right that `device_total_num` naming should be used?

Comment (ilyachur) [reply src/inference/include/openvino/runtime/intel_myriad/hddl_properties.hpp:78]: I hope so

File: src/inference/include/openvino/runtime/intel_myriad/hddl_properties.hpp [hunk 5]
```diff
@@ -0,0 +1,157 @@
...
+ * The number of devices which will be used for stream-affinity must be specified in hddl_service.config file.
+ * Example:
+ * "service_settings":
+ * {
+ *     "stream_device_number":5
+ * }
+ * It means that 5 device will be used for stream-affinity
+ */
+static constexpr Property<std::string, PropertyMutability::RW> setStreamId{"HDDL_SET_STREAM_ID"};
+
```
Comment (ilya-lavrenov) [inline src/inference/include/openvino/runtime/intel_myriad/hddl_properties.hpp:97]: missed properties:
- HDDL_GRAPH_TAG
-

File: src/inference/include/openvino/runtime/intel_myriad/myriad_properties.hpp [hunk 6]
```diff
@@ -0,0 +1,120 @@
...
+ * @brief The flag to reset stalled devices.
+ */
+static constexpr Property<bool, PropertyMutability::RW> enableForceReset{"MYRIAD_ENABLE_FORCE_RESET"};
+
+enum class DDRType {
+    MYRIAD_DDR_AUTO = 0,
+    MYRIAD_DDR_MICRON_2GB,
+    MYRIAD_DDR_SAMSUNG_2GB,
+    MYRIAD_DDR_HYNIX_2GB,
+    MYRIAD_DDR_MICRON_1GB
```
Comment (ilya-lavrenov) [inline src/inference/include/openvino/runtime/intel_myriad/myriad_properties.hpp:74]: add documentation for enum and values

File: src/inference/include/openvino/runtime/intel_myriad/hddl_properties.hpp (hunk 4 above)
Comment (KorA6) [reply src/inference/include/openvino/runtime/intel_myriad/hddl_properties.hpp:78]: Just to clarify:
Do I need to change all the property names from 'camelCase' to 'snake_case'?

File: src/inference/include/openvino/runtime/intel_myriad/hddl_properties.hpp (hunk 3 above)
Comment (KorA6) [reply src/inference/include/openvino/runtime/intel_myriad/hddl_properties.hpp]: Changed

File: src/inference/include/openvino/runtime/intel_myriad/hddl_properties.hpp (hunk 5 above)
Comment (KorA6) [reply src/inference/include/openvino/runtime/intel_myriad/hddl_properties.hpp:115]: Sure, thanks!
Changed

File: src/inference/include/openvino/runtime/intel_myriad/myriad_properties.hpp (hunk 6 above)
Comment (KorA6) [reply src/inference/include/openvino/runtime/intel_myriad/myriad_properties.hpp:83]: Added

File: src/inference/include/openvino/runtime/intel_myriad/hddl_properties.hpp (hunk 4 above)
Comment (KorA6) [reply src/inference/include/openvino/runtime/intel_myriad/hddl_properties.hpp]: Renamed to snake_case

File: src/inference/include/openvino/runtime/intel_myriad/myriad_properties.hpp (hunk 6 above)
Comment (ilya-lavrenov) [reply src/inference/include/openvino/runtime/intel_myriad/myriad_properties.hpp:83]: I don't see..

File: src/inference/include/openvino/runtime/intel_myriad/myriad_properties.hpp [hunk 7]
```diff
@@ -0,0 +1,130 @@
...
+ * If layer is present in such an xml, it would be used during inference even if the layer is natively supported
+ */
+static constexpr Property<std::string, PropertyMutability::RW> custom_layers{"MYRIAD_CUSTOM_LAYERS"};
+
+}  // namespace common
+
+/**
+ * @brief Enum to define possible device protocols
+ */
+enum class Protocol { PCIE = 0, USB };
```
Comment (ilya-lavrenov) [inline src/inference/include/openvino/runtime/intel_myriad/myriad_properties.hpp:36]: document enum values

File: src/inference/include/openvino/runtime/intel_myriad/myriad_properties.hpp (hunk 6 above)
Comment (KorA6) [reply src/inference/include/openvino/runtime/intel_myriad/myriad_properties.hpp]: Now it's definitely changed

File: src/inference/include/openvino/runtime/intel_myriad/myriad_properties.hpp (hunk 7 above)
Comment (KorA6) [reply src/inference/include/openvino/runtime/intel_myriad/myriad_properties.hpp]: Changed

File: src/inference/include/openvino/runtime/intel_myriad/myriad_properties.hpp [hunk 8]
```diff
@@ -0,0 +1,133 @@
...
+//
+
+#pragma once
+
+#include <openvino/runtime/properties.hpp>
+#include <string>
+
+namespace ov {
+namespace intel_myriad {
+namespace common {
```
Comment (ilya-lavrenov) [inline src/inference/include/openvino/runtime/intel_myriad/myriad_properties.hpp:12]: do we need a special namespace `common`? Maybe we can keep all properties in the `intel_myriad` directly..

Comment (KorA6) [reply src/inference/include/openvino/runtime/intel_myriad/myriad_properties.hpp]: Changed

**Comment by KorA6** [comment]
@DariaMityagina @ilya-lavrenov @AntonDudchenko 
Take a look once again please

**Review by ilya-lavrenov — APPROVED** [review]

**Review by DariaMityagina — APPROVED** [review]

**Review by AntonDudchenko — APPROVED** [review]

**Comment by azhogov** [comment]
ci passed: https://github.com/intel-innersource/frameworks.ai.openvino.ci.product-configs/pull/831

**Pull request merged**